### ISSUE TITLE
[C Interop] Cannot use type alias in function

### ISSUE THREAD

**Issue by duarteroso** [vlang/v#22586]
### Describe the bug

I have the following types:
```v
@[typedef]
pub struct C.Camera3D {
}

// Camera, defines position/orientation in 3d space
pub type Camera3D = C.Camera3D

// Camera pub type fallback, defaults to Camera3D
pub type Camera = C.Camera3D
```

Then I have a function defined as follows:
```v
// Update camera position for selected mode
pub fn C.UpdateCamera(camera &Camera, mode int)

// Update camera movement/rotation
pub fn C.UpdateCameraPro(camera &Camera, movement Vector3, rotation Vector3, zoom f32)
```

The `UpdateCamera` function is used as follows:
```v
mut camera := &raylib.Camera3D{ ... }
C.UpdateCamera(camera, int(raylib.CameraMode.camera_free))
```

When calling the `UpdateCamera` function, the following error is thrown:
```bash
error: cannot use `&raylib.Camera3D` as `&raylib.Camera` in argument 1 to `C.UpdateCamera`
   39 |     //
   40 |     for C.WindowShouldClose() == false {
   41 |         C.UpdateCamera(camera, int(raylib.CameraMode.camera_free))
      |                        ~~~~~~
   42 |         //
   43 |         if C.IsKeyDown(int(raylib.KeyboardKey.key_z)) {
```

### Reproduction Steps

1. Get raylib binding from [https://github.com/duarteroso/raylib](https://github.com/duarteroso/raylib)
2. Move to the `examples` branch
3. Run `v run examples/3d_camera_free.v`

The error should be thrown then.

### Expected Behavior

Type alias should be able to resolve to the concrete type.

### Current Behavior

Chaining type aliases doesn't seem to be resolved properly.

### V version

0.4.8

### Environment details (OS name and version, etc.)

V full version: V 0.4.8 5ec9bb5
OS: macos, macOS, 14.7, 23H124
Processor: 16 cpus, 64bit, little endian, Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz

getwd: /Users/duarteroso/.vmodules/raylib
vexe: /Users/duarteroso/code/vlang/v/v
vexe mtime: 2024-10-18 22:24:18

vroot: OK, value: /Users/duarteroso/code/vlang/v
VMODULES: OK, value: /Users/duarteroso/.vmodules
VTMP: OK, value: /tmp/v_501

Git version: git version 2.33.0
Git vroot status: weekly.2024.42-22-g5ec9bb54 (3 commit(s) behind V master)
.git/config present: true

CC version: Apple clang version 15.0.0 (clang-1500.3.9.4)
thirdparty/tcc status: thirdparty-macos-amd64 975f1ad8
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

**Comment by felipensp** [comment]
Can you try again with V `master` version?

**Comment by felipensp** [comment]
Just confirmed, it was fixed by PR #22572
